### PR TITLE
dbft: upgrade dbft library to v0.2.0

### PR DESCRIPTION
**Pull request by AnnaShaleva**

### CODE REVIEW

File: consensus/dbft/block.go [hunk 1]
```diff
@@ -3,17 +3,15 @@ package dbft
 import (
 	"errors"
 	"fmt"
-	
+
```
Comment (roman-khimov) [inline consensus/dbft/block.go:6]: Uh-oh.

File: go.mod [hunk 2]
```diff
@@ -50,21 +50,21 @@ require (
 	github.com/mattn/go-colorable v0.1.13
 	github.com/mattn/go-isatty v0.0.19
 	github.com/naoina/toml v0.1.2-0.20170918210437-9fafd6967416
-	github.com/nspcc-dev/dbft v0.0.0-20240130131324-6d59675ad370
+	github.com/nspcc-dev/dbft v0.2.0
 	github.com/nspcc-dev/neo-go v0.103.1
```
Comment (roman-khimov) [inline go.mod]: Why do we still have it?

File: consensus/dbft/dbft.go [hunk 3]
```diff
@@ -462,21 +458,38 @@ func New(config *params.DBFTConfig, _ ethdb.Database) (*DBFT, error) {
...
+			copy(res.SignatureExt[:], sig)
+			return res
+		}),
+		dbft.WithNewPrepareResponse[common.Hash](func(prepH common.Hash) dbft.PrepareResponse[common.Hash] {
+			res := new(prepareResponse)
+			res.PreparationHashExt = prepH
+			return res
+		}),
+		dbft.WithNewChangeView[common.Hash](func(newView byte, reason dbft.ChangeViewReason, ts uint64) dbft.ChangeView {
+			res := new(changeView)
```
Comment (roman-khimov) [inline consensus/dbft/dbft.go]: Probably would be easier as
```
return &changeView {
    newViewNumber: newView,
...
}
```

File: consensus/dbft/block.go (hunk 1 above)
Comment (AnnaShaleva) [reply consensus/dbft/block.go:6]: Fixed.

File: go.mod (hunk 2 above)
Comment (AnnaShaleva) [reply go.mod]: Because of unfinished PR, fixed.

File: consensus/dbft/dbft.go (hunk 3 above)
Comment (AnnaShaleva) [reply consensus/dbft/dbft.go]: Fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**